core/src/main/java/org/aerogear/mobile/core/api/SdkCore.java

```diff
@@ -0,0 +1,74 @@
+package org.aerogear.mobile.core.api;
+
+
+import android.content.Context;
+
+import org.aerogear.mobile.core.configuration.MobileCoreJsonParser;
+import org.aerogear.mobile.core.configuration.ServiceConfig;
+import org.aerogear.mobile.core.configuration.ServiceConfiguration;
+import org.aerogear.mobile.core.executor.AppExecutors;
+import org.aerogear.mobile.core.http.NetworkApi;
+import org.aerogear.mobile.core.logging.Logger;
+import org.json.JSONException;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Map;
+import java.util.concurrent.Executor;
+
+public class SdkCore implements Core {
```

Comment:
Ideally this class should not have any code and it should just provide implementations for core services.

`getNetworkLayer` can be renamed to provideNetworkLayer etc. and return network interface. 
The same with configuration but to simplify this I have made implementations directly in this class